src/jsonschema/compile_evaluate.cc

```diff
@@ -169,8 +179,16 @@ class EvaluationContext {
     assert(step.relative_instance_location.size() <= 1);
     this->frame_sizes.emplace_back(step.relative_schema_location.size(),
                                    step.relative_instance_location.size());
-    this->evaluate_path_.push_back(step.relative_schema_location);
-    this->instance_location_.push_back(step.relative_instance_location);
+    Pointer relative_schema_location;
+    CONVERT_WEAK_TO_POINTER(relative_schema_location,
+                            step.relative_schema_location);
+    this->evaluate_path_.push_back(relative_schema_location);
+
+    Pointer relative_instance_location;
+    CONVERT_WEAK_TO_POINTER(relative_instance_location,
+                            step.relative_instance_location);
+    this->instance_location_.push_back(relative_instance_location);
+
```

Comment:
Steps or (`SchemaCompilerTemplate`) seems to be coupled to the compiler itself, this why I opt for this conversion step.

Reply:
This could be the culprit for slowness!

Reply:
Yep. See my other comment. I think what we need is a `WeakPointer::push_back` overload that takes `Pointer`